madara/crates/client/gateway/client/src/lib.rs

```diff
@@ -1,3 +1,226 @@
+//!
+//! Returns the public key associated to this node.
+//!
+//! ```json
+//! {
+//!   "jsonrpc": "2.0",
+//!   "method": "get_public_key",
+//!   "id": 1
+//! }
+//! ```
+//!
+//! # Gateway
+//!
+//! The **Gateway**, which you will sometimes see referred to as _gw_ (but still most often as just
+//! _gateway_), is an unspecified endpoint of a Starknet node similar to the [fgw]. It is used by
+//! block producers to receive transactions which are forwarded to it from full nodes which are
+//! unable to produce state locally themselves.
+//!
+//! Historically, this is due to the fact that Starknet used to have a single sequencer responsible
+//! for state production. In practice, this is still very handy for load balancing by having some
+//! ingress nodes receive user queries and validate them before sending them to a sequencer node for
+//! block production. This is because validation, along with block production, both incur a
+//! significant amount of computation and so having them take place on different machines can help
+//! with avoiding bottlenecks in the validation pipeline affecting block production and therefore the
+//! liveliness of the chain.
+//!
+//! This crate implements the methods and deserialization logic to connect to a gateway endpoint
+//! during block sync. For a local implementation of the gateway, check out the gateway server under
+//! `gateway/serve`.
+//!
+//! ## Available endpoints
+//!
+//! Below is a list of endpoints made available to the gateway. Keep in mind that the actual
+//! implementation is subject to subtle yet meaningful differences between nodes, and this includes
+//! the official Starknet gateway endpoints. Some additional, madara-specific methods have also been
+//! added, and it should not be expected that such methods be made available under other node
+//! softwares or even work the same if they are.
+//!
+//! ### `add_transactions`
+//!
+//! An open endpoint for adding transactions forwarded from other full nodes. This includes
+//! [declare transactions], [deploy account transactions] and [invoke transactions].
+//!
+//! ```json5
+//! {
+//!   "jsonrpc": "2.0",
+//!   "method": "add_transactions",
+//!   "params": {
+//!      // Transaction info goes here
+//!   },
+//!   "id": 1
+//! }
+//! ```
+//!
+//! ### `trusted_add_validated_transactions`
+//!
+//! > **This endpoint is unique to Madara.**
+//!
+//! A _trusted_ enpoint, **disabled by default**. This allows adding transactions to the state
+//! without verifying their validity. As discussed above, this can be useful when splitting up the
+//! tasks of validating transactions and adding them to the block production between two nodes.
+//!
+//! ```json5
+//! {
+//!   "jsonrpc": "2.0",
+//!   "method": "trusted_add_validated_transactions",
+//!   "params": {
+//!      // Transaction info goes here
+//!   },
+//!   "id": 1
+//! }
+//! ```
+//!
+//! [`legacy`]: mp_class::ContractClass::Legacy
+//! [`sierra`]: mp_class::ContractClass::Legacy
```

Comment:
not sure what's right but it must be something else
```suggestion
//! [`sierra`]: mp_class::ContractClass::Sierra
```